sbi/samplers/vi/vi_quality_controll.py

```diff
@@ -0,0 +1,352 @@
+from pkg_resources import Distribution
+import torch
+import numpy as np
+
+from typing import Optional, Tuple, Callable
+
+from torch.distributions import constraints
+from torch.distributions import Uniform
+from torch.distributions.transformed_distribution import TransformedDistribution
+from torch.distributions.transforms import AffineTransform, ExpTransform, PowerTransform
+from torch.distributions.utils import broadcast_all
+
+import warnings
+
+from sbi.inference.posteriors.base_posterior import NeuralPosterior
+
+
+_QUALITY_METRIC = {}
+_METRIC_MESSAGE = {}
+
+
+def register_quality_metric(
+    cls: Optional[Callable] = None,
+    name: Optional[str] = None,
+    msg: Optional[str] = None,
+) -> Callable:
+    """This method will register a given metric for cheap quality evaluation of
+    variational posteriors!
+
+    Args:
+        cls: Function to compute the metric.
+        name: Associated name.
+        msg: Short description on how to interpret the metric.
+
+    """
+
+    def _register(cls: Callable) -> Callable:
+        if name is None:
+            cls_name = cls.__name__
+        else:
+            cls_name = name
+        if cls_name in _QUALITY_METRIC:
+            raise ValueError(f"The metric {cls_name} is already registered")
+        else:
+            _QUALITY_METRIC[cls_name] = cls
+            _METRIC_MESSAGE[cls_name] = msg
+        return cls
+
+    if cls is None:
+        return _register
+    else:
+        return _register(cls)
+
+
+def get_quality_metric(name: str) -> Tuple[Callable, str]:
+    """Returns the quality, metric as well as a short description."""
+    return _QUALITY_METRIC[name], _METRIC_MESSAGE[name]
+
+
+def gpdfit(
```

Comment:
Maybe move to `utils`? It's used by quality control and fkl

Reply:
To vi_utils or to sbi.utils?

Reply:
to vi_utils